pkg/apiserver/auditing/backend.go

```diff
@@ -0,0 +1,118 @@
+package auditing
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"k8s.io/klog"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+)
+
+const (
+	WaitTimeout = time.Second
+	WebhookURL  = "https://kube-auditing-webhook-svc.kubesphere-logging-system.svc:443"
+)
+
+type Backend struct {
+	channelCapacity int
+	semCh           chan interface{}
+	cache           chan *EventList
+	client          http.Client
+	sendTimeout     time.Duration
+	waitTimeout     time.Duration
+}
+
+func NewBackend(channelCapacity int, cache chan *EventList, sendTimeout time.Duration) *Backend {
+
+	b := Backend{
+		semCh:           make(chan interface{}, channelCapacity),
+		channelCapacity: channelCapacity,
+		waitTimeout:     WaitTimeout,
+		cache:           cache,
+		sendTimeout:     sendTimeout,
+	}
+
+	b.client = http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+		Timeout: b.sendTimeout,
+	}
+
+	go b.worker()
+
+	return &b
+}
+
+func (b *Backend) worker() {
+
+	// Stop when receiver signal Interrupt.
+	stopCh := b.SetupSignalHandler()
```

Comment:
we've already created `SetupSignalHandler` during apiserver setup, trying to use that one.
https://github.com/kubesphere/kubesphere/blob/master/cmd/ks-apiserver/app/server.go#L78